tests/test_web_app_handler.py

```diff
@@ -109,9 +110,13 @@ def test_invalid_signature(monkeypatch, webhook_event):
 def test_handle_event_with_secret(monkeypatch, webhook_event):
     monkeypatch.setenv("GH_WEBHOOK_SECRET", "fake_secret")
     handler = WebAppHandler(dry_run=True)
+    handler.gh_client.send_trace = MagicMock()
```

Comment:
I would use `with mock.patch.object(handler.gh_client, 'send_trace') as mck:` that way it is undone (this'll also appease mypy when we add it since mypy forbids assigning to methods)